xc-admin/packages/xc-admin-common/src/multisig_transaction/WormholeInstruction.ts

```diff
@@ -0,0 +1,50 @@
+import { createReadOnlyWormholeProgramInterface } from "@certusone/wormhole-sdk/lib/cjs/solana/wormhole";
+import { WormholeInstructionCoder } from "@certusone/wormhole-sdk/lib/cjs/solana/wormhole/coder/instruction";
+import { getPythClusterApiUrl } from "@pythnetwork/client/lib/cluster";
+import { Connection, TransactionInstruction } from "@solana/web3.js";
+import { MultisigInstruction } from ".";
+import { decodeGovernancePayload } from "../governance_payload";
+import { AnchorAccounts, resolveAccountNames } from "./anchor";
+
+export class WormholeInstruction implements MultisigInstruction {
+  readonly program = "Wormhole Bridge";
+  readonly name: string;
+  readonly args: { [key: string]: any };
+  readonly accounts: AnchorAccounts;
+
+  constructor(instruction: TransactionInstruction) {
```

Comment:
Make a constructor that simply takes the needed private members and stores them, and put this logic in a static factory method like `WormholeInstruction.fromTransactionInstruction()`. 

(This is a general engineering tip. If you do any complex logic in the constructor, it becomes harder for people to instantiate objects for tests etc. Your code is more flexible if you have a simple constructor combined with factory methods around it.)

Reply:
Yes